test/engine/SpatialJoinAlgorithmsTest.cpp

```diff
@@ -780,8 +781,277 @@ TEST_P(SpatialJoinParamTest, computeResultSmallDatasetDifferentSizeChildren) {
       expectedMaxDist10000000_rows_diff, columnNames, false);
 }
 
-INSTANTIATE_TEST_SUITE_P(SpatialJoin, SpatialJoinParamTest, ::testing::Bool());
+INSTANTIATE_TEST_SUITE_P(
+    SpatialJoin, SpatialJoinParamTest,
+    ::testing::Values(SpatialJoin::Algorithm::Baseline,
+                      SpatialJoin::Algorithm::S2Geometry,
+                      SpatialJoin::Algorithm::BoundingBox));
 
 }  // end of Namespace computeResultTest
 
+namespace boundingBox {
+
+namespace bg = boost::geometry;
+namespace bgi = boost::geometry::index;
+
+typedef bg::model::point<double, 2, bg::cs::cartesian> point;
+typedef bg::model::box<point> box;
+typedef std::pair<point, size_t> value;
+
+inline void testBoundingBox(const long long& maxDistInMeters,
+                            const point& startPoint) {
+  auto checkOutside = [&](const point& point1, const point& startPoint,
+                          const std::vector<box>& bbox,
+                          SpatialJoinAlgorithms* spatialJoinAlg) {
+    // check if the point is contained in any bounding box
+    bool within = spatialJoinAlg->OnlyForTestingWrapperContainedInBoundingBoxes(
+        bbox, point1);
+    if (!within) {
+      GeoPoint geo1{point1.get<1>(), point1.get<0>()};
+      GeoPoint geo2{startPoint.get<1>(), startPoint.get<0>()};
+      double dist = ad_utility::detail::wktDistImpl(geo1, geo2) * 1000;
+      ASSERT_GT(static_cast<long long>(dist), maxDistInMeters);
+    }
+  };
+
+  auto testBounds = [](double x, double y, const box& bbox,
+                       bool shouldBeWithin) {
+    // correct lon bounds if necessary
+    if (x < -180) {
+      x += 360;
+    } else if (x > 180) {
+      x -= 360;
+    }
+
+    // testing only possible, if lat bounds are correct and the lon bounds
+    // don't cover everything (as then left or right of the box is again
+    // inside the box because of the spherical geometry)
```

Comment:
And what do we have to check in that case? Or can't we do anything?

Reply:
if we have a bounding box, which goes from -180 to 180 longitude, then left of the bounding box is just in the bounding box again. (i.e. -180.00001 is the same as +179.99999). As all longitudes are covered, a left or right bound does not exist (on the sphere this makes intuitive sense). A test in that case is not necessary, because this part of the code is testing, if the points, which are barely not in the bounding box are correctly identified as not being in the bounding box.

I added the above explanation to the code